parso/python/errors.py

```diff
@@ -961,12 +975,26 @@ def _check_assignment(self, node, is_deletion=False, is_namedexpr=False):
               or '_test' in type_
               or type_ in ('term', 'factor')):
             error = 'operator'
+        elif type_ == 'name':
+            if node.value == '__debug__':  # should be a keyword for all versions
+                if (2, 7) < self._normalizer.version < (3, 8):
+                    error = 'keyword'
+                else:
+                    error = str(node.value)
+                not_debug = False
+                is_keyword = True
```

Comment:
IMO this belongs to `_NameChecks`, because there we already check for `__debug__`.

Reply:
It might be worth patching is_definition.

Reply:
Just patched, I suppose it's working fine :)